---
title: "Figure 1.- Defining rDNA haplotypes in the C57BL/6J strain"
---

## 0.- Setup

```{r, echo=F, message=F}
the_data_fd <- '../../data/'
source('../../scripts/setup.R')
source('../../scripts/haplotypes.R')
```

## A.- Ultralong Nanopore: Allele frequencies 

For this analysis we rely solely on the default mode of Megalodon. This is,
we do not allow for supplementary alignments. Although this reduces the 
amount of data we can utilise, and prevent us from analysing co-location,
it increases the reliability of our calls to establish the haplotypes to
study and their associated positions.

```{r}
variants_mef_all <- read_tsv(
  '../../data/mef/no_supp/per_read_variant_calls.txt', col_types=cols()) %>%
  mutate(pos=pos+1,
         pos_ref=adjust_pos((pos %% 45306), 3008)) %>%
  select(read_id, pos_ref, pos, ref=ref_seq, ref_prob=ref_log_prob,
         alt=alt_seq, alt_prob=alt_log_prob) %>%
  mutate(ref_prob=exp(ref_prob), alt_prob=exp(alt_prob)) %>%
  pivot_longer(c(ref_prob, alt_prob), values_to='prob', names_to='dominant') %>%
  filter(prob>0.9) %>%
  group_by(read_id, pos, pos_ref, ref, alt) %>%
  filter(n()==1) %>%
  mutate(dominant=gsub('_prob', '', dominant),
         allele=if_else(dominant=="ref", ref, alt)) %>%
  ungroup() %>%
  select(read_id, pos_ref, allele, dominant) %>%
  group_by(read_id) %>%
  filter(n_distinct(pos_ref)==n()) %>%
  ungroup()
```


```{r}
linkage_mef_all <- variants_mef_all %>%
  inner_join(variants_mef_all,
             by=c('read_id'),
             suffix=c('_1', '_2')) %>%
  filter(pos_ref_1 < pos_ref_2) %>% 
  group_by(pos_ref_1, dominant_1,
           pos_ref_2, dominant_2) %>%
  summarise(num_reads=n_distinct(read_id), .groups='drop') %>%
  mutate(haplotype=paste(dominant_1, dominant_2, sep='_')) %>%
  pivot_wider(-c(dominant_1, dominant_2),
              names_from='haplotype', values_from='num_reads', values_fill=0) %>%
  rowwise() %>%
  mutate(
    broom::tidy(
      fisher.test(
        matrix(c(ref_ref, ref_alt, alt_ref, alt_alt),
           byrow=T, nrow=2)))) %>%
  ungroup()
```

```{r}
all_linked_pos <- c(-104, linkage_mef_all %>%
    filter(pos_ref_1 %in% c(-104, 12736) | pos_ref_2 %in% c(-104, 12736)) %>%
    mutate(linked_pos=if_else(
        pos_ref_1 %in% c(-104, 12736), pos_ref_2, pos_ref_1)) %>%
    select(linked_pos, everything(),
           -pos_ref_1, -pos_ref_2, -method, -alternative) %>%
    filter(p.value < 1e-6) %>% pull(linked_pos) %>% unique())
```

```{r}
the_haplotypes <- variants_mef_all %>%
    filter(pos_ref %in% c(-104, 1305, 8063, 12736)) %>%
    select(-dominant) %>%
    group_by(read_id) %>%
    filter(n()==4, n_distinct(pos_ref)==n()) %>%
    pivot_wider(names_from='pos_ref', values_from='allele') %>%
    unite(haplotype) %>%
    separate(haplotype, sep='_',
             into=c('read_id', 'haplotype_ext'),
             extra='merge') %>%
    group_by(haplotype_ext) %>%
    filter(n() > 5)
```


```{r}
haplotype_variants_mef_all <- variants_mef_all %>%
  inner_join(the_haplotypes, by=c('read_id')) %>%
  filter(pos_ref %in% all_linked_pos) %>%
  filter(pos_ref > -1000) %>%
  group_by(pos_ref, haplotype_ext) %>%
  mutate(total_count=n()) %>%
  group_by(pos_ref, allele, dominant, haplotype_ext) %>%
  summarise(freq=n()/min(total_count), .groups='drop') %>%
  group_by(pos_ref, haplotype_ext) %>%
  filter(! any(n_distinct(allele)==2 & min(freq) > 0.2)) %>%
  group_by(pos_ref) %>%
  filter(n_distinct(haplotype_ext)==4) %>%
  group_by(pos_ref, haplotype_ext) %>%
  filter(freq==max(freq)) %>%
  group_by(pos_ref, allele)
```

```{r}
haplotype_variants_mef_all_p <- haplotype_variants_mef_all %>%
  mutate(haplotype=adjust_haplo(haplotype_ext)) %>%
  select(-haplotype_ext) %>%
  group_by(pos_ref, allele) %>%
  mutate(
      haplo_unique=case_when(
          n_distinct(haplotype) == 1  ~ 'Subhaplotype',
          n_distinct(haplotype) == 2 ~ '-104 Haplotype',
          TRUE ~ 'Non-specific')) %>%
  ungroup() %>%
  bind_rows(tribble(~pos_ref, ~haplotype, ~freq, ~haplo_unique, ~allele,
                    0, 'A_T_A', 0, 'Non-specific', 'A',
                    0, 'A_T_G', 0, 'Non-specific', 'A',
                    0, 'C_C_A', 0, 'Non-specific', 'A',
                    0, 'C_T_A', 0, 'Non-specific', 'A')) %>%
  group_by(haplo_unique) %>%
  mutate(haplo_unique=factor(haplo_unique,
                             levels=c('Subhaplotype',
                                      '-104 Haplotype',
                                      'Non-specific'))) %>%
  ungroup() %>%
  arrange(pos_ref) %>%
  group_by(pos_ref) %>%
  mutate(face=case_when(
      (pos_ref >= coding_regions_m[1,]$start &
         pos_ref <= coding_regions_m[1,]$end) |
          (pos_ref >= coding_regions_m[2,]$start &
             pos_ref <= coding_regions_m[2,]$end) |
          (pos_ref >= coding_regions_m[3,]$start &
             pos_ref <= coding_regions_m[3,]$end) ~ 'bold',
      TRUE ~ 'plain')) %>%
  ungroup() %>%
  mutate(pos_ref=as.factor(pos_ref),
         pos_ref=fct_recode(pos_ref, TSS='0'))
```

```{r}
haplotype_alleles_df <- haplotype_variants_mef_all_p %>%
  filter(haplo_unique != 'Non-specific') %>%
  separate(haplotype, into=c('allele_104'), extra='drop', remove = F) %>%
  mutate(haplotype=if_else(haplo_unique=='-104 Haplotype',
                           allele_104, haplotype)) %>%
  group_by(pos_ref, allele) %>%
  summarise(haplotype=unique(haplotype), .groups='drop') %>%
  mutate(pos_ref=as.numeric(as.character(pos_ref)))
```



```{r warning=F}
faces <- haplotype_variants_mef_all_p %>%
  select(pos_ref, face) %>% unique() %>% pull(face)
haplotype_variants_mef_p <- haplotype_variants_mef_all_p %>% 
  mutate(haplotype=gsub('_', '', haplotype)) %>%
  mutate(haplo_unique=fct_recode(
    haplo_unique,
    'Haplotype' = 'Subhaplotype',
    'A/C'='-104 Haplotype')) %>%
  ggplot() +
  geom_col(aes(x=pos_ref, y=freq,
               fill=allele,
               colour=haplo_unique,
               alpha=haplo_unique,
               linetype=haplo_unique)) +
  geom_vline(xintercept='TSS', alpha=0.3) +
  theme_bw() +
  facet_grid(rows=vars(haplotype)) +
  labs(x='', y='Allele Frequency') +
  scale_fill_manual(name='Allele', values=allele_colours_cb) +
  scale_colour_manual(name='Specificity',
                      values=c('Haplotype'='black',
                               'A/C'='grey30',
                               'Non-specific'='white')) +
  scale_alpha_manual(name='Specificity',
                     values=c('Haplotype'=0.9,
                              'A/C'=0.9,
                              'Non-specific'=0.4)) +
  scale_linetype_manual(name='Specificity',
                        values=c('Haplotype'='solid',
                                 'A/C'='blank',
                                 'Non-specific'='blank')) +
  scale_y_continuous(breaks=c(0.5, 1)) +
  theme(axis.text.x=element_text(
    angle=90, hjust=1, vjust=0.5, face=faces, size=12),
    axis.text.y=element_text(size=12),
    axis.title=element_text(size=14),
    panel.grid = element_blank(),
    panel.border=element_rect(size=1.1),
    strip.background=element_blank(),
    strip.text.y=element_text(angle=0, hjust=0, size=16),
    legend.position='bottom',
    legend.title=element_text(size=12, face='bold'),
    legend.text=element_text(size=12),
    legend.margin=margin(t=-20, r=0, b=0, l=0))
```

```{r}
haplotype_cov_p <- read_tsv('../../data/strains/wgs/covs_rdna/S15.cov',
         col_types=cols(),
         col_names=c('chr', 'pos', 'cov')) %>%
  mutate(pos_ref=adjust_pos(pos)) %>%
  filter(pos_ref >= -1000, pos_ref <= 13500 ) %>%
  mutate(cov_norm=cov/sum(cov)) %>%
  ggplot() +
  geom_vline(xintercept=0, alpha=0.3) +
  geom_rect(data=coding_regions_m,
            aes(xmin=start, xmax=end, ymin=-Inf, ymax=Inf),
            fill='grey50', alpha=0.1) +
  geom_ribbon(aes(ymin=0, ymax=cov_norm, x=pos_ref)) +
  scale_x_continuous(limits=c(-1000, 13500),
                     breaks=seq(from=0, to=12500, by=2500),
                     labels=c('TSS',
                              seq(from=2500, to=12500, by=2500))) +
  theme_minimal() +
  labs(x='', y='') +
  theme(panel.grid=element_blank(),
        axis.text.y=element_blank(),
        axis.text.x=element_blank(),
        legend.title=element_text(face='bold'),
        legend.margin=margin(t=-5, r=0, b=0, l=0),
        axis.text=element_text(size=13),
        axis.title=element_text(size=13))
```


```{r}
haplotype_pos_p <- haplotype_alleles_df %>%
  mutate(haplotype=gsub('_', '', haplotype)) %>%
  ggplot() +
  geom_point(aes(x=pos_ref, y=0), shape=23, size=1, alpha=0.9) +
  geom_rect(data=coding_regions_m,
            aes(xmin=start, xmax=end, ymin=-Inf, ymax=Inf),
            fill='grey50', alpha=0.1) +
  geom_vline(xintercept=0, alpha=0.3) +
  theme_minimal() +
  scale_fill_manual(name='Allele',
                    values=allele_colours_cb) +
  scale_colour_manual(name='Allele',
                      values=allele_colours_cb) +
  scale_y_continuous(limits=c(-0.1, 0.1), breaks=c(0)) +
  scale_x_continuous(limits=c(-1000, 13500),
                     breaks=seq(from=0, to=12500, by=2500),
                     labels=c('TSS',
                              seq(from=2500, to=12500, by=2500))) +
  theme(strip.text.y.left=element_text(angle=0, hjust=1, size=13),
        panel.grid.major.x=element_blank(),
        panel.grid.minor.x=element_blank(),
        panel.grid.minor.y=element_blank(),
        panel.spacing=unit(0, 'lines'),
        axis.text.y=element_blank(),
        legend.title=element_text(face='bold'),
        legend.margin=margin(t=-5, r=0, b=0, l=0),
        axis.text=element_text(size=13),
        axis.title=element_text(size=13)) +
  labs(x='Position in rDNA unit', y='')
```


```{r}
fig_1b_fd <- '../../figures/accepted/fig_1/'
dir.create(fig_1b_fd, showWarnings = FALSE, recursive = T)

ggsave(paste0(fig_1b_fd, 'B_haplotype_pos.pdf'),
       haplotype_pos_p, height=1 , width=12, dpi=1200)
ggsave(paste0(fig_1b_fd, 'B_haplotype_cov.pdf'),
       haplotype_cov_p, height=1 , width=12, dpi=1200)
ggsave(paste0(fig_1b_fd, 'B_haplotype_allele_freqs.pdf'),
       haplotype_variants_mef_p, height=3 , width=12, dpi=1200)
```


## B.- Ultralong Nanopore: Co-localisation frequency 

```{r}
mef_variants_supp <- read_tsv('../../data/mef/supp/per_read_variant_calls.txt',
                     col_types=cols()) %>%
  filter(chrm=="BK000964.3_looped_3008_20_units") %>%
  select(-chrm) %>%
  mutate(pos=pos+1,
         unit_num=as.integer( ( pos - 1) / 45306) + 1,
         pos_ref=adjust_pos((pos %% 45306), 3008)) %>%
  group_by(read_id, strand, ref_log_prob, alt_log_prob,
           ref_seq, alt_seq, var_id, pos_ref) %>%
  summarise(unit_num=min(unit_num), .groups='drop') %>%
  ungroup()
```

```{r}
mef_haplotypes_supp <- mef_variants_supp %>%
  filter(pos_ref %in% c(-104, 1305, 8063, 12736)) %>%
  mutate(ref_prob=exp(ref_log_prob), alt_prob=exp(alt_log_prob)) %>%
  pivot_longer(c(ref_prob, alt_prob), values_to='prob', names_to='dominant') %>%
  filter(prob>0.9) %>%
  mutate(allele=if_else(dominant=="ref_prob", ref_seq, alt_seq)) %>%
  group_by(read_id, unit_num) %>%
  mutate(num_entries=n(), num_pos=n_distinct(pos_ref)) %>%
  filter(num_entries==4, num_pos==4) %>%
  ungroup() %>%
  select(read_id, pos_ref, allele, unit_num) %>%
  pivot_wider(names_from=pos_ref, values_from=allele) %>%
  relocate(c(`-104`, `1305`, `8063`, `12736`), .after=unit_num) %>%
  unite(haplotype) %>%
  separate(haplotype, sep='_', into=c('read_id', 'unit_num', 'haplotype'),
           extra='merge') %>%
  mutate(unit_num=as.numeric(unit_num)) %>%
  group_by(haplotype) %>%
  mutate(haplotype=adjust_haplo(haplotype)) %>%
  ungroup()
```


```{r}
mef_haplotypes_co  <- mef_haplotypes_supp %>% 
  group_by(read_id, haplotype) %>%
  summarise(num_units=n(), .groups='drop') %>%
  group_by(read_id) %>%
  filter(sum(num_units) > 1, all(haplotype %in% names(haplotype_colours))) %>%
  pivot_wider(names_from=haplotype, values_from=num_units, values_fill=0) %>%
  ungroup()
```

```{r}
mef_haplotypes_co_freqs <- 
  inner_join(mef_haplotypes_co %>%
             pivot_longer(
               -read_id, names_to='haplotype', values_to='num_units'), 
         mef_haplotypes_co, by='read_id') %>%
  filter(num_units > 0) %>%
  group_by(read_id) %>%
  mutate(num_units=sum(num_units)) %>%
  ungroup() %>%
  mutate(A_T_A=A_T_A/num_units,
         A_T_G=A_T_G/num_units,
         C_C_A=C_C_A/num_units,
         C_T_A=C_T_A/num_units) %>%
  group_by(haplotype) %>%
  pivot_longer(-c(read_id, haplotype, num_units),
               names_to='haplotype_2', values_to='freq') %>%
  group_by(haplotype, haplotype_2) %>%
  summarise(freq=mean(freq), .groups='drop') %>%
  ungroup()
```

```{r}
mef_haplotypes_co_p <- mef_haplotypes_co_freqs %>%
  mutate(haplotype=gsub('_', '', haplotype),
         haplotype_2=gsub('_', '', haplotype_2)) %>%
  ggplot() +
  geom_tile(aes(x=haplotype, y=haplotype_2, fill=freq),
            colour='black', size=1.1) +
  theme_bw() +
  labs(x='', y='') +
  scale_fill_gradient2(name='', low='white', high='grey20',
                       guide = guide_legend(reverse = T),
                       limits = c(0,1)) +
  theme(axis.text=element_text(size=20),
        legend.key.height=unit(1.5, 'cm'), 
        legend.text=element_text(size=20),
        panel.border=element_blank(),
        panel.grid=element_blank(),
        axis.ticks=element_blank()) +
  coord_fixed()
```


```{r}
dir.create(fig_1c_fd, showWarnings = FALSE, recursive = T)
ggsave('../../figures/accepted/fig_1/C_haplotypes_colocation.pdf',
       mef_haplotypes_co_p, height=5, width=6, dpi=1200)
```


## Supplementary

```{r}
fig_1b_supp_fd <- '../../figures/publication/fig_1/B/supp/'
```

### Linkage

```{r}
linkage_mef_p <- D
```


```{r}
dir.create(fig_1b_supp_fd, showWarnings = FALSE, recursive = T)
ggsave(paste0(fig_1b_supp_fd, 'linkage_mef.pdf'),
       linkage_mef_p, height=4.2, width=4.2, dpi=1200)
```

### Haplotypes alleles

```{r}
haplotype_alleles_p <- haplotype_alleles_df %>%
  mutate(haplotype=gsub('_', '', haplotype)) %>%
  ggplot() +
  geom_rect(aes(xmin=pos_ref-5, xmax=pos_ref+5, ymin=-0.1, ymax=0.1,
                fill=allele, colour=allele), size=0.1, alpha=0.9) +
  geom_rect(data=coding_regions_m,
            aes(xmin=start, xmax=end, ymin=-Inf, ymax=Inf),
            fill='grey50', alpha=0.1) +
  geom_vline(xintercept=0, alpha=0.3) +
  facet_grid(rows=vars(haplotype),
             switch='y') +
  theme_minimal() +
  scale_fill_manual(name='Allele',
                    values=allele_colours_cb) +
  scale_colour_manual(name='Allele',
                    values=allele_colours_cb) +
  scale_y_continuous(limits=c(-0.15, 0.15), breaks=c(0)) +
  scale_x_continuous(limits=c(-1000, 13500),
                     breaks=seq(from=0, to=12500, by=2500),
                     labels=c('TSS', seq(from=2500, to=12500, by=2500))) +
  theme(strip.text.y.left=element_text(angle=0, hjust=1, size=13),
        panel.grid.major.x=element_blank(),
        panel.grid.minor.x=element_blank(),
        panel.grid.minor.y=element_blank(),
        panel.spacing=unit(0, 'lines'),
        axis.text.y=element_blank(),
        legend.title=element_text(face='bold'),
        legend.margin=margin(t=-5, r=0, b=0, l=0),
        axis.text=element_text(size=13),
        axis.title=element_text(size=13)) +
  labs(x='Position in rDNA unit', y='')
```

```{r}
dir.create(fig_1b_supp_fd, showWarnings = FALSE, recursive = T)
ggsave(paste0(fig_1b_supp_fd, 'haplotypes_alleles.pdf'),
       haplotype_alleles_p, height=2, width=12, dpi=1200)
```


### Haplotype Frequencies

```{r}
haplotype_freqs_mef <- the_haplotypes %>%
  rename(haplotype=haplotype_ext) %>%
  mutate(haplotype=adjust_haplo(haplotype),
         haplotype=gsub('_', '', haplotype)) %>%
  group_by(haplotype) %>%
  summarise(num_reads=n(), .groups='drop') %>%
  ungroup() %>%
  mutate(freq_reads=num_reads/sum(num_reads))
```

```{r}
haplotype_freqs_mef_supp <- mef_haplotypes_supp %>%
  mutate(haplotype=adjust_haplo(haplotype),
         haplotype=gsub('_', '', haplotype)) %>%
  filter(nchar(haplotype)==3) %>%
  group_by(haplotype) %>%
  summarise(num_reads=n(), .groups='drop') %>%
  ungroup() %>%
  mutate(freq_reads=num_reads/sum(num_reads))
```

```{r}
haplotype_freqs_mef_df <- bind_rows(
  haplotype_freqs_mef %>% mutate(supp=F),
  haplotype_freqs_mef_supp %>% mutate(supp=T)) %>%
  mutate(the_label=if_else(supp,'Supplementary alignments enabled',
                           'No supplementary alignments'))
```

```{r}
haplotype_freqs_mef_p <- haplotype_freqs_mef_df %>% 
  ggplot() +
  geom_col(aes(x=haplotype, y=freq_reads, fill=haplotype),
           colour='black', size=0.7, alpha=0.8) +
  geom_text(aes(label=the_label),
            x=2.5, y=0.45, size=4,
            data=(haplotype_freqs_mef_df %>%
              select(supp, the_label) %>%
              distinct())) +
  facet_grid(rows=vars(supp)) +
  scale_y_continuous(limits=c(0, 0.5),
                     breaks=c(0, 0.25, 0.5),
                     labels=c('', 0.25, 0.5)) +
  scale_fill_manual('', values=haplotype_colours) +
  theme_bw() +
  labs(x='', y='Frequency') +
  theme(legend.position='none',
        panel.grid=element_blank(),
        panel.border=element_rect(size=1.1),
        panel.spacing=unit(0.2, 'lines'),
        strip.background=element_blank(),
        strip.text=element_blank(),
        axis.text=element_text(size=12),
        axis.title=element_text(size=14))
```

```{r}
ggsave('../../figures/publication/fig_1/B/supp/haplotype_freqs_mef.pdf',
       haplotype_freqs_mef_p, height=4 , width=4, dpi=1200)
```

